Emby.Server.Implementations/Library/UserDataManager.cs

```diff
@@ -28,25 +28,24 @@ public class UserDataManager : IUserDataManager
 
         private readonly ILogger _logger;
         private readonly IServerConfigurationManager _config;
-
-        private Func<IUserManager> _userManager;
-
-        public UserDataManager(ILoggerFactory loggerFactory, IServerConfigurationManager config, Func<IUserManager> userManager)
+        private readonly IUserManager _userManager;
+        private readonly IUserDataRepository _repository;
+
+        public UserDataManager(
+            ILogger<UserDataManager> logger,
+            IServerConfigurationManager config,
+            IUserManager userManager,
+            IUserDataRepository repository)
         {
+            _logger = logger;
             _config = config;
-            _logger = loggerFactory.CreateLogger(GetType().Name);
             _userManager = userManager;
+            _repository = repository;
         }
 
-        /// <summary>
-        /// Gets or sets the repository.
-        /// </summary>
-        /// <value>The repository.</value>
-        public IUserDataRepository Repository { get; set; }
```

Comment:
you're removing a public method here, are you sure no plugin or anything uses it?

Reply:
Hmm no, I hadn't considered that plugins might be using it. Plugins should really be getting this service via dependency injection instead of using a public property like this.

It should be an easy fix in plugins that were relying on this property, but I can revert it if you think it would be better to keep it.

Reply:
Okay, this is going to 10.6 which would presumably break plugins anyway... :) so this is not a blocker.